test/conformance/api/v1/revision_timeout_test.go

```diff
@@ -100,134 +84,122 @@ func TestRevisionTimeout(t *testing.T) {
-		t.Errorf("Failed request with sleep 0s with revision timeout 5s: %v", err)
-	}
-
-	// Fail by surpassing the initial timeout.
-	if err := sendRequest(t, clients, rev2sURL, 5*time.Second, 0, http.StatusGatewayTimeout); err != nil {
-		t.Errorf("Did not fail request with sleep 5s with revision timeout 2s: %v", err)
-	}
-	if err := sendRequest(t, clients, rev5sURL, 7*time.Second, 0, http.StatusGatewayTimeout); err != nil {
-		t.Errorf("Did not fail request with sleep 7s with revision timeout 5s: %v", err)
-	}
+	testCases := []struct {
+		name           string
+		shouldScaleTo0 bool
+		timeoutSeconds int64
+		expectedStatus int
+		initialSleep   int
+		sleep          int
+	}{{
+		name:           "when scaling up from 0 and does not exceed timeout seconds",
+		shouldScaleTo0: true,
+		timeoutSeconds: 10,
+		expectedStatus: http.StatusOK,
+		initialSleep:   0,
+		sleep:          0,
+	}, {
+		name:           "when scaling up from 0 and it writes first byte before timeout",
+		shouldScaleTo0: true,
+		timeoutSeconds: 10,
+		expectedStatus: http.StatusOK,
+		initialSleep:   0,
+		sleep:          15,
+	}, {
+		name:           "when scaling up from 0 and it does exceed timeout seconds",
+		shouldScaleTo0: true,
+		timeoutSeconds: 1, // If the pods come up faster than 1s, this test might fail.
+		expectedStatus: http.StatusGatewayTimeout,
+	}, {
+		name:           "when pods already exist, and it does not exceed timeout seconds",
+		timeoutSeconds: 10,
+		expectedStatus: http.StatusOK,
+		initialSleep:   2,
+		sleep:          0,
+	}, {
+		name:           "when pods already exist, and it does exceed timeout seconds",
+		timeoutSeconds: 10,
+		expectedStatus: http.StatusGatewayTimeout,
+		initialSleep:   12,
+		sleep:          0,
+	}, {
+		name:           "when pods already exist, and it writes first byte before timeout",
+		timeoutSeconds: 10,
+		expectedStatus: http.StatusOK,
+		initialSleep:   0,
+		sleep:          15,
+	}}
 
-	// Not fail by not surpassing in the initial timeout, but in the overall request duration.
-	if err := sendRequest(t, clients, rev2sURL, time.Second, 3*time.Second, http.StatusOK); err != nil {
-		t.Errorf("Did not fail request with sleep 1s/3s with revision timeout 2s: %v", err)
-	}
-	if err := sendRequest(t, clients, rev5sURL, 3*time.Second, 3*time.Second, http.StatusOK); err != nil {
-		t.Errorf("Failed request with sleep 3s/3s with revision timeout 5s: %v", err)
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			names := test.ResourceNames{
+				Service: test.ObjectNameForTest(t),
+				Image:   test.Timeout,
+			}
+
+			test.CleanupOnInterrupt(func() { test.TearDown(clients, names) })
+			defer test.TearDown(clients, names)
+
+			t.Log("Creating a new Service ")
+			svc, err := createService(t, clients, names, tc.timeoutSeconds)
+			if err != nil {
+				t.Fatal("Failed to create Service:", err)
+			}
+
+			names.Route = serviceresourcenames.Route(svc)
```

Comment:
Are these being used?

Reply:
They were there from the previous version, looks like they were never actually required. Removed.